app/helpers/language_helper.py

```diff
@@ -30,6 +31,11 @@ def handle_language(metadata: Optional[MetadataProxy] = None) -> None:
         g.allowed_languages = get_allowed_languages(schema_name, launch_language)
         request_language = request.args.get("language_code")
         if request_language and request_language in g.allowed_languages:
+            if metadata:
+                schema = load_schema_from_metadata(
+                    metadata=metadata, language_code=request_language
+                )
+                cookie_session["survey_title"] = schema.json["title"]
```

Comment:
The latest flask login changes mean the cookie is only sent when the cookie session is modified, therefore to avoid sending the cookie when it hasn't changed, i.e. page refresh etc, the title in the cookie session should only be set if it doesn't match what is set in there. A cookie is sent even if you set the value to the same existing value.

```suggestion
                if schema.json["title"] != cookie_session.get("survey_title"):
                    cookie_session["survey_title"] = schema.json["title"]
```